### ISSUE TITLE
Allow a tag to be displayed next to operation name in search view

### ISSUE THREAD

**Issue by bboreham** [jaegertracing/jaeger-ui#709]
## Requirement - what kind of business use case are you trying to solve?

Presented with a set of traces in the search view, I would like to pick ones of interest more quickly, by having more information available on each row.

Example how this might look:

![image](https://user-images.githubusercontent.com/8125524/108878048-c22c7f00-75f7-11eb-8d6b-9941175d1f82.png)

This idea is very similar to #487, but about traces in the search view rather than spans in the trace view.

## Problem - what in Jaeger blocks you from solving the requirement?

In current Jaeger I can add filters into the Tags field, but typing those out not conducive to exploration.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

It could be a template like that implemented for #487, but I think it should go further and allow different options for different kinds of span.

## Any open questions to address

Haven't figured out a good syntax for the template yet.  Maybe like awk?
```
/HTTP GET/ {#http.url}
/grpc/ {#process.hostname}
```
In extremis, this template becomes a DSL, ref https://github.com/jaegertracing/jaeger/issues/1811

I also feel this could be processed server-side, as an extension of #247, rather than make the browser do all the matching.

**Comment by yurishkuro** [comment]
I feel like this has to be more dynamic than a static config per installation. Essentially, this sounds more like we need a feature extraction pipeline that can represent traces in tabular format and an analytical storage that allows querying / filtering / group-bys on some standard columns.

**Comment by bboreham** [comment]
> more dynamic

Now you mention it, before I started hacking on the code my thought was that as a user I should be able to configure by point-and-click, "when you see this kind of trace, pull the URL tag up to the search view".

> an analytical storage that allows querying / filtering / group-bys

Is that a progression of the GraphQL thoughts that have been posted here, or something different?
I'm a little daunted by the size of that.

**Comment by yurishkuro** [comment]
> Is that a progression of the GraphQL thoughts that have been posted here, or something different? I'm a little daunted by the size of that.

No, unrelated to GraphQL. Yes, it is a very large scope. It is feasible to do if the storage backend supports such queries (Elasticsearch does).

What you propose is a simpler version, I guess. Not opposed to it, but also not sure how useful it will be, especially because the data set is limited to the search results. Dynamic filtering of search results is also a sizable scope.
